app/models/timer.rb

```diff
@@ -0,0 +1,19 @@
+class Timer < ApplicationRecord
+  belongs_to :product
+  has_many :autions
+
+  validates :start_at, presence: true
+  validates :period, presence: true
+  validates :bid_step, presence: true
+  validates :status, presence: true
+  validate :time_validation
+
+  def time_validation
+    if self[:end_at] < self[:start_at]
```

Comment:
`if end_at < start_at` not use self

Reply:
Tạo folder `validators` viết tất cả những validate custom vào trong folder đó. Sửa lại mấy cái validate đã viết trước đây vào trong folder validators
Tham khảo: https://guides.rubyonrails.org/active_record_validations.html#custom-validators